CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+## How to contribute code
+
+Follow these steps to submit your code contribution.
+
+### Step 1. Open an issue
+
+Before making any changes, we recommend opening an issue (if one doesn't already
+exist) and discussing your proposed changes. This way, we can give you feedback
+and validate the proposed changes.
+
+If the changes are minor (simple bug fix or documentation fix), then feel free
+to open a PR without discussion.
+
+### Step 2. Make code changes
+
+To make code changes, you need to fork the repository. You will need to setup a
+development environment and run the unit tests. This is covered in section
+"Setup environment".
+
+### Step 3. Create a pull request
+
+Once the change is ready, open a pull request from your branch in your fork to
+the master branch in [keras-team/keras](https://github.com/keras-team/keras).
+
+### Step 4. Sign the Contributor License Agreement
+
+After creating the pull request, the `google-cla` bot will comment on your pull
+request with instructions on signing the Contributor License Agreement (CLA) if
+you haven't done so. Please follow the instructions to sign the CLA. A `cla:yes`
+tag is then added to the pull request.
+
+![Tag added](https://i.imgur.com/LHEdIfL.png)
+
+
+### Step 5. Code review
+
+CI tests will automatically be run directly on your pull request.  Their
+status will be reported back via GitHub actions.
+
+There may be
+several rounds of comments and code changes before the pull request gets
+approved by the reviewer.
+
+![Approval from reviewer](https://i.imgur.com/zgRziTt.png)
+
+### Step 6. Merging
+
+Once the pull request is approved, a `ready to pull` tag will be added to the
+pull request. A team member will take care of the merging.
+
+![Ready to pull](https://i.imgur.com/yCEqJsA.png)
+
+Here is an [example pull request](https://github.com/keras-team/keras/pull/15015)
+for your reference.
+
+## Setup environment
+
+Setting up your KerasCV development environment is quite easy.  You simply
+need to run the following commands:
+
+```shell
+git clone https://github.com/YOUR_GITHUB_USERNAME/keras.git
+cd keras
+pip install ".[tests]"
```

Comment:
what is ".[test]"? Usually we list all the required dependency in a requirements.txt file, and use pip install -r requirements.txt.

Reply:
.[tests] installs everything required to run unit tests.  You can see how this is handled in KerasTuner as well:
https://github.com/keras-team/keras-tuner/blob/master/setup.py#L40

So, installing:
```
pip install .
``` 
will install the list on line 30.  `pip install .[tests]` will also install pytest, flake8, isort, etc.

WDYT?

Reply:
+ @haifeng-jin  + @mattdangerw to discuss best practices for managing dependencies as well.

I personally am a fan of the `setup.py` dependency management system.

Reply:
I am not sure! I don't really know enough about what is common practice in the python packaging world (or is this a legitimate source of contention?). It is nice to be able to split the testing and package requirements.

Some discussion:
https://stackoverflow.com/q/43658870
https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/

`install_requires` works and we have it now, so I am fine to leave as is unless we have a good reason to switch.

Reply:
Some extra context: the benefit of requirements is you can pin versions.

But at Google we are forced to update all dependencies anyways, so for Keras I think install_requires is superior.

Reply:
If requirements.txt makes it easier to pin a system installed tensorflow that might be a reason to use it. I am not sure, but for running cloud TPU jobs with the setup.py we have today, I've needed to hack it up to stop it from installing a new tf version that does not have proper TPU support. No need to resolve here, but that's something I will keep looking into.

Reply:
I would suggest using `setup.py`. One caveat is that the current setting for version number [[Link](https://github.com/keras-team/autokeras/blob/master/setup.cfg#L1-L2)] may throw an error when installing with `pip install .` because it imports the package to get the version number before installing anything, which may encounter some not installed packages. For example, the user would have to install tensorflow before running `pip install .`. It is not a big deal though. The user can tell which package to install from the error.